ls1intum/Artemis
# `Development`: Remove LtiUserId association to user

### Checklist
#### General
#### Server

### Motivation and Context
The **ltiUserId** table associated the id of an Artemis user with the id of user in the external LMS (e.g Moodle). This was problematic because we could have users from different LMS's with the same user id in their original LMS, causing them to wrongly be mapped to the same user in Artemis. Additionally, this table wasn't really necessary since it was only used to check if a user was an LTI user on initialization, which we can instead just do by checking if the user belong to the "LTI" group, which is added to all users created during an LTI launch.

### Description
All logic associated to the ltiUserId table is removed. The table itself will be dropped in a follow-up PR. On initialization of a user, to check if a user is an LTI user, instead of using the ltiUserId table, now we check if a user belongs to the "LTI" group.

### Steps for Testing
Prerequisites:
- 1 Instructor in Artemis
- 1 Instructor in an external LMS (https://moodle.ase.in.tum.de/my/courses.php)

1. Setup an online course in Artemis as the instructor
2. Create an exercise for the course
3. Setup an LTI exercise in the external LMS
4. Verify that launching as a new user to Artemis works, you get a generated password, and you can then use that password to login
5. Verify that grading the new user returns a grade to the external LMS
6. Verify that launching as an existing LTI user still works, and that grading also works for these users


### Review Progress

#### Code Review
#### Manual Tests

### Test Coverage
| Class/File | Branch | Line |
|------------|-------:|-----:|
| UserResource.java | 72% | 84% |
| LtiService.java | 100% | 99% |
| Lti10Service.java | 84% | 95% |
| Lti13Service.java | 93% | 95% |
